Fix: Apply ellipsis for overflowing content.

VANSH3104 opened the pull request:
<!-- Describe your pull request here.-->

Fixes: <!-- Issue link, or clear description.-->
fix issue #31392  with reference to the past pr #33091
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.
change CSS for ellipsis 
Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![Screenshot from 2025-01-25 00-04-03](https://github.com/user-attachments/assets/c82f81ba-fa85-4ea0-bd48-83c2182dd4d5)
![Screenshot from 2025-01-25 00-03-52](https://github.com/user-attachments/assets/3a66a900-ebce-4b88-9998-2c3a3f770919)
![Screenshot from 2025-01-25 00-03-41](https://github.com/user-attachments/assets/8aea5b80-f994-43c7-aa5c-4c6a7bcd0879)
![Screenshot from 2025-01-25 00-03-29](https://github.com/user-attachments/assets/b6e7081a-e1cb-4758-8ced-d1d922630236)
![Screenshot from 2025-01-25 00-03-24](https://github.com/user-attachments/assets/cde10dac-e295-4473-9d0c-7577daf2a2ea)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

VANSH3104 commented:
Hey @alya, sorry for the inconvenience. I ran into a Git issue that caused my PR to be closed, so I need to make an another pr. I’ve also minimized the code and improved it using CSS. Let me know if you need anything else!

VANSH3104 commented:
Hey @amaan784, sorry for the inconvenience. I’ve adjusted the code and made the changes in CSS as you requested. I’ve also removed the JS logic. Let me know if you need anything else!

alya commented:
OK, please link to the previous PR for reference, so that this one can be reviewed.

alya commented:
Also, please update your commit message to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages), and complete the self-review checklist.

VANSH3104 commented:
@alya i gave the reference to my past pr

VANSH3104 commented:
@alya I also corrected my commit message according to the manual and did self-review checks

alya commented:
Please explain how you addressed @amanagr 's feedback at https://github.com/zulip/zulip/pull/33091#issuecomment-2612590010 .

VANSH3104 commented:
@alya I have addressed @amanagr  feedback in #33091 by replacing the hardcoded JavaScript approach with a more CSS-driven solution. The necessary changes have been implemented using CSS to handle styling and interactions wherever possible, reducing reliance on JavaScript. Let me know if any further adjustments are needed

amanagr commented on web/styles/typeahead.css:
```diff
@@ -136,6 +136,9 @@
            icon alignment and spacing, when icons
            are shown. */
+        overflow: hidden;
+        max-width: calc(25ch);
```
We don't want to hardcode 25 character limit here. Also this code changes should be above the comment.

VANSH3104 commented:
Hi @amanagr, I've added a max width to ensure that the content doesn't stretch too wide, as this causes issues with displaying other elements like pronouns. The truncation with the ellipsis (...) effect enhances the UI, making it more consistent and user-friendly. I’ve also updated the corresponding .hbs files to ensure this works seamlessly. Could you please take a look at the changes? Let me know if you have any suggestions or concerns!
![Screenshot from 2025-02-04 15-17-55](https://github.com/user-attachments/assets/14d6237c-1f33-4941-b3c0-54aa006db8ae)

amanagr commented:
Can you open  a separate PR just fixing the space issue between `#` and stream name?

![image](https://github.com/user-attachments/assets/5044fb19-41b0-486c-ab24-156ca4182463)

amanagr commented on web/styles/zulip.css:
```diff
@@ -1956,7 +1956,8 @@ body:not(.hide-left-sidebar) {
     gap: 4px;
     font-weight: 400;
     line-height: var(--base-line-height-unitless);
-    white-space: normal;
+    white-space: nowrap;
+    overflow: hidden;
```
This looks like unrelated change.

VANSH3104 replied:
yes this is unrelated i will remove it

amanagr commented on web/styles/typeahead.css:
```diff
@@ -144,6 +145,13 @@
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        max-width: 200px;
```
I think we should only apply a `max-width` if the typeahead doesn't have any of the secondary text.

VANSH3104 replied:
yes you are right i will remove this too

VANSH3104 commented on web/styles/zulip.css:
```diff
@@ -1969,6 +1970,12 @@ body:not(.hide-left-sidebar) {
         margin-top: 2px;
     }
 
+    .stream-name {
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+    }
+
```
I applied these changes to achieve the ellipsis (...) effect for better UI enhancement. If I remove it, the design appears less structured 
![Screenshot from 2025-02-06 13-31-48](https://github.com/user-attachments/assets/690dab4c-ab7b-4d9a-b0ed-f66713d39c29)
after adding this it look like 
![Screenshot from 2025-02-06 13-35-08](https://github.com/user-attachments/assets/f51b11b3-8534-4d90-a8ed-ce6f0ba9eef1)
and here white-space is not needed I remove it also

VANSH3104 replied:
Should I make changes to this, update my PR, or keep it as it is? as i use overflow-hidden to make the stream-name in between the div and then convert overflow into ellipsis as described in this issue whats your thoughts ?

amanagr replied:
Yeah, this looks correct. Can you open a separate PR for that since this change is not related to the issue of typeaheads?

VANSH3104 commented:
> Can you open a separate PR just fixing the space issue between `#` and stream name?
> 
> ![image](https://private-user-images.githubusercontent.com/25124304/409995395-5044fb19-41b0-486c-ab24-156ca4182463.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg4Mzc1NTIsIm5iZiI6MTczODgzNzI1MiwicGF0aCI6Ii8yNTEyNDMwNC80MDk5OTUzOTUtNTA0NGZiMTktNDFiMC00ODZjLWFiMjQtMTU2Y2E0MTgyNDYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjA2VDEwMjA1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NGYxZjFiOWMxYTk2ZmRjZjkyYWJkNDFjMzVmZjIwOTlhZTFjYjRiNzc5ZTYzY2ZlNTU1NjBmYzVhODFiZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aBVt3PZlXi-Qa5TOG0BnnG83E_Ug3hVloopOKptzX-8)
i solved it here #33369

VANSH3104 commented:
@amanagr I updated the code and removed all the unwanted CSS add-ons can you review my code ?

VANSH3104 commented:
@amanagr  any updates on this any thing more needed in this ?

amanagr commented:
@VANSH3104 it is hard to review your changes since you are removing the changes you added in your first commit. Can you fix that?

amanagr commented:
I think this solution is incorrect and we want to abbreviate the whole typeahead item here instead of just the primary text which is incorrectly assumed to be just stream name in this PR. Closing this as we are close to the release now, so I will open a PR for this.
